demos/python_demos/asl_recognition_demo/asl_recognition_demo.py

```diff
@@ -0,0 +1,175 @@
+                      default=None)
+    args.add_argument('--no_show', action='store_true',
+                      help='Optional. Do not visualize inference results.')
+
+    return parser
+
+
+def load_class_map(file_path):
+    """ Returns class names map. """
+
+    if file_path is not None and exists(file_path):
+        with open(file_path, 'r') as input_stream:
+            data = json.load(input_stream)
+            if isinstance(data, dict):
+                keys = list(data)
+                assert len(keys) > 0
+
+                if isinstance(keys[0], str):
+                    class_map = {v: k for k, v in data.items()}
+                elif isinstance(keys[0], int):
+                    class_map = data
+                else:
+                    raise ValueError('Unknown format of the class map has been provided.')
+            elif isinstance(data, list):
+                class_map = dict(enumerate(data))
+            else:
+                raise ValueError('Unknown data type: {}'.format(type(data)))
+
+    else:
+        class_map = None
+
+    return class_map
+
+
+def main():
+    """ Main function. """
+
+    log.basicConfig(format='[ %(levelname)s ] %(message)s', level=log.INFO, stream=sys.stdout)
+    args = build_argparser().parse_args()
+
+    ie_core = load_ie_core(args.device, args.cpu_extension)
+
+    person_detector = PersonDetector(args.detection_model, args.device, ie_core,
+                                     num_requests=2, output_name=DETECTOR_OUTPUT_NAME)
+    action_recognizer = ActionRecognizer(args.action_model, args.device, ie_core,
+                                         num_requests=2, img_scale=ACTION_IMAGE_SCALE,
+                                         num_classes=ACTION_NUM_CLASSES)
+
+    video_stream = VideoStream(args.input, ACTION_NET_INPUT_FPS, action_recognizer.input_length)
+    video_stream.start()
+
+    person_tracker = PersonTracker(person_detector, TRACKER_SCORE_THRESHOLD, TRACKER_IOU_THRESHOLD)
+
+    class_map = load_class_map(args.class_map)
+    assert class_map is not None
+
+    last_caption = None
+    person_roi = None
+
+    start_time = time.perf_counter()
+    while True:
+        frame = video_stream.get_live_frame()
+        batch = video_stream.get_batch()
+
+        person_roi = person_tracker.get_roi(frame)
+        if person_roi is not None:
+            recognizer_result = action_recognizer(batch, person_roi)
+            if recognizer_result is not None:
+                action_class_id = np.argmax(recognizer_result)
+                action_class_label = \
+                    class_map[action_class_id] if class_map is not None else action_class_id
+
+                action_class_score = np.max(recognizer_result)
+                if action_class_score > ACTION_SCORE_THRESHOLD:
+                    last_caption = 'Last: {} '.format(action_class_label)
+
+        if args.no_show:
+            continue
+
```

Comment:
These lines should be placed just before `cv2.imshow('Demo', frame)` to minimize the difference of what the demo does

Reply:
Fixed.